Azure/sonic-buildimage
# [sonic-cfggen][FRR] Support T2 chassis frontend

**- What I did**

Support new role: T2 chassis frontend switch (SpineChassisFrontendRouter). The chassis frontend switch needs BGP-EVPN, Vxlan and Vnet support.
 
**- How I did it**

* Modify minigraph parser to add Vnet and Vxlan informration for T2 chassis frontend

* Add new bgpd and zebra j2 templates to support BGP-EVPN 

* Add new tests to sonic-cfggen unit test scripts

**- How to verify it**

Run load_minigraph and verify entries in config DB and FRR configuration.

In config DB, you should be able to see the following entries:

```
"VNET|VnetFE"
"VXLAN_TUNNEL|TunnelInt"
```
In FRR docker, you should be able to see BGP-EVPN configuration like follows:
```
router bgp 4000 vrf VnetFE
  no bgp default ipv4-unicast
  bgp log-neighbor-changes
  bgp bestpath as-path multipath-relax
  no bgp default ipv4-unicast
  bgp graceful-restart restart-time 240
  bgp graceful-restart
  bgp router-id 4.0.0.0
  neighbor 192.168.0.1 remote-as 3000
  neighbor 192.168.0.1 description Leaf01
  neighbor 192.168.0.1 timers 3 10
  address-family ipv4 unicast
    neighbor 192.168.0.1 activate
    neighbor 192.168.0.1 soft-reconfiguration inbound
    maximum-paths 64
  exit-address-family
  address-family l2vpn evpn
    advertise ipv4 unicast
  exit-address-family

!
! bgp multiple-instance
!
route-map FROM_BGP_SPEAKER_V4 permit 10
!
route-map TO_BGP_SPEAKER_V4 deny 10
!
router bgp 4000
  bgp log-neighbor-changes
  bgp bestpath as-path multipath-relax
  no bgp default ipv4-unicast
  bgp graceful-restart restart-time 240
  bgp graceful-restart
  bgp router-id 4.0.0.0
  network 4.0.0.0/32
  neighbor 4.0.0.1 remote-as 4000
  neighbor 4.0.0.1 description SpineFront02
  neighbor 4.0.0.1 timers 3 10
  address-family l2vpn evpn
    neighbor 4.0.0.1 activate
    advertise-all-vni
  exit-address-family
  neighbor 172.16.0.2 remote-as 5000
  neighbor 172.16.0.2 description SpineBack01
  neighbor 172.16.0.2 timers 3 10
  address-family ipv4 unicast
    neighbor 172.16.0.2 allowas-in 1
    neighbor 172.16.0.2 activate
    neighbor 172.16.0.2 soft-reconfiguration inbound
    maximum-paths 64
  exit-address-family
  neighbor 172.16.0.10 remote-as 5000
  neighbor 172.16.0.10 description SpineBack02
  neighbor 172.16.0.10 timers 3 10
  address-family ipv4 unicast
    neighbor 172.16.0.10 allowas-in 1
    neighbor 172.16.0.10 activate
    neighbor 172.16.0.10 soft-reconfiguration inbound
    maximum-paths 64
  exit-address-family
!
maximum-paths 64
!
route-map ISOLATE permit 10
set as-path prepend 4000
!
route-map set-next-hop-global-v6 permit 10
set ipv6 next-hop prefer-global
!
```

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Wei Bai <baiwei0427@gmail.com> and Ze Gan <ganze718@gmail.com>
